content/en/runtime.md

```diff
@@ -0,0 +1,44 @@
+---
+title: Runtime
+status: Feedback Appreciated
+category: concept
+tags: ["application", "", ""]
+---
+
+## What it is
+
+A runtime, in general, is something that executes a piece of software.
+It is an abstraction of the underlying operating system that knows how to translate the program's commands into respective actions on the operating system level. 
+
+In the context of [cloud native](/cloud-native-apps/), we refer to the container runtime when we talk about the _runtime_. 
+A container runtime specifically implements the [Open Container Initiative](https://opencontainers.org/) specification to ensure consistent handling around the ecosystem. 
+This includes the container image specification, the method by which the runtime can retrieve an image, 
+and how that image is unpacked, layered, mounted, and executed. 
+These runtimes utilize kernel features such as namespaces and cgroups to provide isolation and resource management for the containers.
+
+## Problem it addresses
+
+Container runtimes are a necessary component of container orchestrators such as Kubernetes. 
+The runtime is scoped to only handle the lifecycle of the containers. 
+User-facing features, such as scaling and service discovery, fault tolerance, etc., are managed by the orchestrator or some other high-level tool that interfaces with the runtime.
+
+Without a container runtime, the user would have to manually handle the container lifecycle, interface directly with cgroups and namespaces in the kernel, and handle hardware resources.
```

Comment:
I'm confused. If a runtime is an abstraction, wouldn't the problem be that the app would have to interface directly with the OS, and we'd need to ensure interoperability for all possible apps with all possible OSs, and that would be a headache?  As far as I understood, abstractions are not to automate things but to enable interoperability. To be clear, I never really new what a runtime is, so this is only based on my understanding of abstraction.

Reply:
@akshatcoder-hash did you already add the simplification?! Instead of throwing in cgroups and namespaces and kernel, we could just say operating system.

Reply:
@akshatcoder-hash did you already add the simplification?! Instead of throwing in cgroups and namespaces and kernel, we could just say operating system.